### PR TITLE
 Updated the ajax to send json body correctly

### PR DESCRIPTION
**Pull request by Phara0h**
Without this, there is no way to send a correct put or post with a json body. If we had to default to one I would say majority of people who use this have a service that uses json instead of old bloated xml.

### CODE REVIEW

**Comment by dinsley** [comment]
+1

**Comment by Phara0h** [comment]
any updates in getting this into a release?

File: template/utils/send_sample_request.js [hunk 1]
```diff
@@ -93,15 +93,51 @@ define([
...
+               ajaxRequest.data = JSON.stringify(param);
+           }
+       }
+
       $.ajax(ajaxRequest);
 
+      function paramToQueryParms(param) {
+        var sendString = [];
+        for (var k in param)
+        {
```
Comment (jeffrey008) [inline template/utils/send_sample_request.js]: The curl brace and for loop should be on the same line. Similar problem to the if/else below.
The coding style should be consistent.

File: template/utils/send_sample_request.js [hunk 2]
```diff
@@ -93,15 +93,51 @@ define([
...
+          if(param[k] != null && param[k] != "")
```
Comment (jeffrey008) [inline template/utils/send_sample_request.js]: Better do 
```js
if(param[k] == null || param[k] == "") {
  continue;
}
```
to avoid nested if cases.

**Comment by Phara0h** [comment]
Oh dang forgot I even posted this, it's been 3 years lol

**Pull request closed without merging**